src/main/scala/terminal/FileSystem.scala

```diff
@@ -0,0 +1,113 @@
+package terminal
+
+import java.util.NoSuchElementException
+
+
+class FileSystem {
+  //key = command;  value = how many paths need to be specified
+  val commandsAndNoOfPaths = Map("pwd" -> 0, "mkdir" -> 1, "ls" -> 0, "h" -> 0, "q" -> 0 )
+
+  var rootPath : Folders = _
+  var currentLevel : Folders = _
+  var cmdPrefix : String = ""
+  var path1_from : String = ""
+  var path2_to : String = ""
+
+
+
+  def run() : Unit = {
+    //initialize the root
+    rootPath = new Folders("")
+    rootPath.currentPath = "/"
+    currentLevel = rootPath
+
+    //run the terminal continuously until is chosen q to quit
+    commandInterpretor()
+  }
+
+
+  def commandInterpretor() : Unit = {
+    print( """Your cmd (press "q" to quit, "h" for help): """ )
+    val keyboardInput = getKeyboardInput
+
+    splitToCmdAndPaths( keyboardInput )
+
+    cmdPrefix match {
+      case "pwd"        => println( getCurrentDirectoryPath ); unitializeCmdAndPaths(); commandInterpretor()
+      case "mkdir"      => if(path1_from.nonEmpty) makeFolder(); unitializeCmdAndPaths(); commandInterpretor()
+      case "ls"         => listAllFilesAndFolders(); unitializeCmdAndPaths(); commandInterpretor()
+      case "h"          => help(); unitializeCmdAndPaths(); commandInterpretor()
+      case "q"          => println("You chose q, the app will close.")
+      case _            => println( "Command not recognized." ); commandInterpretor()     //this default case will keep the "terminal" running even in case if is a command not recognized
+    }
+  }
+
+  def getKeyboardInput : String = scala.io.StdIn.readLine()
+
+  def getCurrentDirectoryPath : String = currentLevel.currentPath
+
+  def help() : Unit = {
+    println(
+      """Can be used the following commands:
+        |pwd
+        |mkdir
+        |ls""".stripMargin)
+  }
+
+  def splitToCmdAndPaths( inputKeyboard : String ) : Unit = {
+    try{
+      //find the command; handle the case if the command does not exist
+      this.cmdPrefix = commandsAndNoOfPaths.filter(
+        x => inputKeyboard.startsWith(x._1)
+      ).keys.head    //return into cmdPrefix the first element (and the only element) of the List which met the condition "inputKeyboard.startsWith( x )"
```

Comment:
Maps are optimized for search by key. Also, there is the possibility of the user giving an unrecognized command, in which case i believe your code will fail. I recommend parsing the user input to extract the command and parameter then do something like `commandsAndNoOfPaths.get(parsedCommand).getOrElse(/*help command*/)`. <- Noticed that i treated the case where the command is unrecognized